### PR TITLE
feat: 28 Files2vue compatibility

### PR DESCRIPTION
**Pull request by skjnldsv**
https://github.com/nextcloud/server/pull/39808

Fixes https://github.com/nextcloud/text/issues/4626

- [x] Once https://github.com/nextcloud/server/pull/39792 is merged
  - [x] forward port https://github.com/nextcloud/text/pull/4680
- [x] Fix cypress tests
  - [x] Uncaught exception on preview loading of folders https://github.com/nextcloud/server/pull/39950
  - [x] Viewer compatibility https://github.com/nextcloud/viewer/pull/1878 
- [ ] Mark remaining failures that take longer as skipped -> follow up ticket
  - [ ] BUG: File list header not shown in a folder. filelist._currentDir is undefined and falls back to `/`
  - [ ] Public share file list still missing (therefore rich workspace and file open tests for that are broken)
  - [ ] New file creation still missing, probably https://github.com/nextcloud/server/pull/39945 but not part of that yet
  - [ ] remove exception handling for https://github.com/nextcloud/server/pull/40065 once that is merged.

[F2V final overview https://github.com/nextcloud/server/issues/39914](https://github.com/nextcloud/server/issues/39914)

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -48,4 +48,12 @@ webpackConfig.resolve.modules = [
 	'node_modules'
 ]
 
+// Fix Buffer issues
+webpackConfig.plugins.push(new webpack.ProvidePlugin({
+	Buffer: ['buffer', 'Buffer'],
+}))
+webpackConfig.resolve.fallback = {
+	buffer: require.resolve('buffer'),
+}
```
Comment (juliushaertl) [inline webpack.config.js:57]: Should this be upstreamed?

Comment (skjnldsv) [reply webpack.config.js:57]: No clue, but it's required somehow
ref https://github.com/nextcloud-libraries/webpack-vue-config/issues/450#issuecomment-1674291089

**Comment by juliushaertl** [comment]
Will take over for finishing tests and adjusting cypress to work with the f2v merge

File: cypress/e2e/nodes/ImageView.spec.js [hunk 2]
```diff
@@ -2,13 +2,9 @@ import { randUser } from '../../utils/index.js'
 
 const user = randUser()
 
-const refresh = () => cy.get('.files-controls .crumb:not(.hidden) a')
-	.last()
-	.click({ force: true })
-
```
Comment (skjnldsv) [inline cypress/e2e/nodes/ImageView.spec.js:8]: You can still refresh by clicking the home icon btw :)

File: cypress/e2e/attachments.spec.js [hunk 3]
```diff
@@ -361,7 +361,7 @@ describe('Test all attachment insertion methods', () => {
 	it('[share] check everything behaves correctly on the share target user side', () => {
 		// check the file list
 		cy.getFile('test.md')
-			.should('contain', 'test.md')
+			.should('contain', 'test .md')
```
[the remark this answers is not in the document]
Comment (max-nextcloud) [reply cypress/e2e/attachments.spec.js:364]: sounds good. That's nothing text needs to worry about.

**Comment by max-nextcloud** [comment]
cypress tests that failed on the last run succeeded for me locally. :shrug:

**Comment by max-nextcloud** [comment]
Follow up ticket:

* [ ] BUG: File list header not shown in a folder. filelist._currentDir is undefined and falls back to `/`
* [ ] Public share file list still missing (therefore rich workspace and file open tests for that are broken)
* [ ] New file creation still missing. we're not even testing that yet. Add some tests once it's there.
* [ ] remove exception handling for https://github.com/nextcloud/server/pull/40065 once that is merged.

**Review by max-nextcloud — APPROVED** [review]

**Pull request merged**